### PR TITLE
Add Developer section with CI subsection and GitHub Actions page

### LINKED ISSUE #423: Document GitHub Actions with GPUs
@betatim wrote a great blog post about how `scikit-learn` set up GPU CI with GitHub Actions Hosted GPU Runners.

https://betatim.github.io/posts/github-action-with-gpu/

We should add a similar section to the deployment docs with the following sections:

- How to set up hosted GPU runners
  - Add credit card
  - Set spending limit
  - Add a new hosted runner pool
  - Choose NVIDIA VM image for runner pool
  - Choose NVIDIA hardware for runner pool
  - Configure workflow to use new GPU runner pool
- Best practices on getting the most from your runners
  - Add label restriction to avoid abuse
  - Run nightly to catch regressions
  - Trigger on certain code paths

### PR DESCRIPTION
**Pull request by jacobtomlinson**
This PR adds a new "Developers" to the documentation. It also adds a "CI" subsection with a "GitHub Actions" page.

The new page covers:

- GPU runners in GitHub Actions
- How to configure them
- How to set a spending cap
- How to configure workflows to use them
- Best practices for getting the most out of your runners
  - Only run on certain labels
  - Run nightly
  - Only run on certain code paths
 

cc @betatim who wrote the original blog post for scikit-learn that inspired this page

Closes #423

### CODE REVIEW

File: source/developer/ci/github-actions.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+![Screenshot of the GitHub Actions runner configuration page with the new runner button highlighted](/_static/images/developer/ci/new-hosted-runner.png)
+
+Next you need to give your runner a name, for example `linux-nvidia-gpu`, you'll need to remember this for configuring your workflows later. Then you need to choose your runner settings:
+
+- Under "Platform" select "Linux x64"
+- Under "Image" switch to the "Partner" tab and choose "NVIDIA GPU-Optimized Image for AI and HPC"
+- Under "Size" switch to the "GPU-powered" tab and select your preferred NVIDIA hardware
+
+![Screenshot of the GitHub Actions runner configuration page a new GPU runner configured](/_static/images/developer/ci/new-runner-config.png)
```
Comment (mmccarty) [inline source/developer/ci/github-actions.md:33]:
```suggestion
![Screenshot of the GitHub Actions runner configuration page a new GPU runner configured](/_static/images/developer/ci/github-actions/new-runner-config.png)
```

File: source/developer/ci/github-actions.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+### Setting up your GPU runners
+
+First you'll need to set up a way to pay GitHub. You can do this by [adding a payment method](https://docs.github.com/en/billing/managing-your-github-billing-settings/adding-or-editing-a-payment-method) to your organisation.
+
+While you're in your billing settings you should also decide what the maximum is that you wish to spend on GPU CI functionality and then set a [spending limit](https://docs.github.com/en/billing/managing-billing-for-github-actions/managing-your-spending-limit-for-github-actions) on your account.
+
+Next you can go into the GitHub Actions settings for your account and configure a [larger runner](https://docs.github.com/en/actions/using-github-hosted-runners/using-larger-runners/about-larger-runners). You can find this settings page by visiting `https://github.com/organizations/<YOUR_ORG>/settings/actions/runners`.
+
+![Screenshot of the GitHub Actions runner configuration page with the new runner button highlighted](/_static/images/developer/ci/new-hosted-runner.png)
```
Comment (mmccarty) [inline source/developer/ci/github-actions.md:25]:
```suggestion
![Screenshot of the GitHub Actions runner configuration page with the new runner button highlighted](/_static/images/developer/ci/github-actions/new-hosted-runner.png)
```

**Review by mmccarty — CHANGES_REQUESTED** [review]
Also need to add this somewhere...
```
checking consistency... /Users/mmccarty/sandbox/rapidsai/deployment/source/developer/index.md: WARNING: document isn't included in any toctree
done
```

**Comment by jacobtomlinson** [comment]
Thanks for the review @mmccarty. Build is passing now.

File: source/index.md [hunk 3]
```diff
@@ -98,4 +98,14 @@ Detailed guides on how to deploy and optimize RAPIDS.
 {bdg}`Infiniband`
 {bdg}`MIG`
 ````
+
+````{grid-item-card}
+:link: developer/index
+:link-type: doc
+{fas}`wrench;sd-text-primary` Guides
```
Comment (mmccarty) [inline source/index.md]:
```suggestion
{fas}`wrench;sd-text-primary` Developer
```

Comment (mmccarty) [reply source/index.md]: Suggesting "Developer"? since we already have a Guides section

**Review by mmccarty — APPROVED** [review]
Looks great, thanks!

**Pull request merged**